Gordon360/Services/RecIM/ParticipantService.cs

```diff
@@ -255,6 +330,22 @@ public bool IsAdmin(string username)
         {
             return _context.Participant.Any(p => p.Username == username && p.IsAdmin == true);
         }
+
+        private string GetCustomUnqiueUsername(string username)
+        {
+            if (_context.Participant.Any((p) => p.Username == username && p.IsCustom == true))
+            {
+                var index = 2;
+                string newUsername = username + index.ToString();
+                while (_context.Participant.Any((p) => p.Username == newUsername && p.IsCustom == true))
+                {
+                    index++;
+                }
+                return newUsername;
```

Comment:
does this logic work? I don't think index is passed by reference here. Most definitely newUsername is not being re-assigned a value.

Reply:
I have tested with `barack.obama` and it does create a `barack.obama2`, I will do more tests with Gordon users.